projects/components/src/skeleton/skeleton.component.scss

```diff
@@ -0,0 +1,114 @@
+@mixin loading-animation {
+  content: '';
+  animation: skeleton-animation 1.2s infinite;
+  height: 100%;
+  left: 0;
+  position: absolute;
+  right: 0;
+  top: 0;
+  transform: translateX(-100%);
+  z-index: 1;
+  background: linear-gradient(90deg, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0.4), rgba(255, 255, 255, 0));
+}
+
+@mixin initial-location {
+  top: initial;
+  left: initial;
+  transform: initial;
+}
+
+.skeleton {
+  position: relative;
+  background-color: #dee2e6;
+  overflow: hidden;
+
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+
+  border-radius: 6px;
+}
+
+.skeleton::after {
+  @include loading-animation;
+}
+
+.skeleton.skeleton-rectangle {
```

Comment:
instead of enumerating all child classes this way, you can do the following
```
.skeleton {
  &.skeleton-rectangle {}
  &.skeleton-rectangle-text {}
  ...
}
```